### PR TITLE
Add design considerations to explainer

### LINKED ISSUE #27: Document the choice of using streams
It would be good to determine and write down the pros and cons of this approach.
This could be done in the spec or directly in this issue.

### PR DESCRIPTION
**Pull request by alvestrand**
In particular, mention the pros and cons of using Streams.

Fixes #27

### CODE REVIEW

**Comment by alvestrand** [comment]
tagging @youennf

**Comment by youennf** [comment]
Looks good as a first version. I am not sure I would mention 'this is complex' since this is subjective. We might also need to envision this for 'opposite-direction APIs' and maybe even for 'same-direction APIs'.

File: explainer.md [hunk 1]
```diff
@@ -246,13 +246,21 @@ There are some challenges with the Streams interface:
...
 adequate for real-time processing; the streaming part is not contributing
 very much to slowing down the pipelines.
 
+## Alternatives to Streams ##
+The alternatives involve callback-based or event-based interfaces; those
+would require developing new interfaces that allow the relevant WebRTC
+objects to be visible in the worker context in order to do processing off
+the main thread. This would seem to be a significantly bigger specification
+and implementation effort.
+
```
Comment (guidou) [inline explainer.md:263]: would it make sense to mention worklets?
In that case, my impression is that the implementation effort was much greater and the API would be more complex, at least if we want it to be as general/flexible as streams.

Comment (alvestrand) [reply explainer.md:263]: I'll add some words about worklets, and then merge.

**Pull request merged**